docs/details/main-components/animation.rst

```diff
@@ -4,107 +4,149 @@
 Animation (lv_anim)
 ===================
 
-You can automatically change the value of a variable between a start and
-an end value using animations. Animation will happen by periodically
-calling an "animator" function with the corresponding value parameter.
-
-The *animator* functions have the following prototype:
+Animations allow you to define the way something should move or change over time, and
+let LVGL do the heavy lifting of making it happen.  What makes it so powerful is that
+the thing being changed can be virtually anything in your system.  It is very
+convenient to apply this to LVGL Widgets in your user interface (UI), to change their
+appearance, size or location over time.  But because it is --- at its core --- a
+generic change-over-time manager, complete with a variety of optional event
+callbacks, its application can be wider than just to UI components.
+
+For each Animation you create, it accomplishes the above by providing a generic
+method of varying a signed integer from a start value to an end value over a
+specified time period.  It allows you to specify what object it applies to (the
+"variable"), which is available in the callback functions that are called as the
+Animation is playing through.
+
+This variation over time can be linear (default), it can be on a path (curve) that
+you specify, and there is even a variety of commonly-used non-linear effects that can
+be specified.
+
+The main callback called during an Animation (when it is playing) is called an
+*animator* function, which has the following prototype:
 
 .. code-block:: c
 
-   void func(void * var, lv_anim_var_t value);
+    void func(void * var, lv_anim_var_t value);
```

Comment:
> This lv_anim_var_t type seems to be gone now, fyi. I think it's fine to leave though? Not sure.

Great catch! 

As this the current prototype we should use this in the docs too.

```suggestion
    void func(void * var, int32_t value);
```

Reply:
> As this the current prototype we should use this in the docs too.

It is already so in the update I pushed Friday.